src/test/scala/chisel3/tests/AsyncResetNativeTest.scala

```diff
@@ -0,0 +1,171 @@
+    if(true) Seq() else Seq(VerilatorBackendAnnotation)
+  }
+
+  "demonstrate how register immediately changes on reset without waiting for clock" in {
+    test(new NativeAsyncResetRegModule(11)).withAnnotations(annotations) { dut =>
+      dut.io.in.poke(7.U)
+      dut.io.out.expect(11.U)
+
+      dut.clock.step()
+      dut.io.out.expect(7.U)
+
+      dut.io.in.poke(8.U)
+      dut.io.out.expect(7.U)
+
+      dut.clock.step()
+
+      dut.io.out.expect(8.U)
+
+      dut.reset.poke(true.B)
+      dut.io.in.poke(3.U)
+      dut.io.out.expect(11.U)
+
+      dut.reset.poke(false.B)
+      dut.io.out.expect(11.U)
+
+      dut.clock.step()
+      dut.io.out.expect(3.U)
+    }
+  }
+  "register is one, after tester startup's default reset" in {
+    test(new NativeAsyncResetRegModule(1)).withAnnotations(annotations) { dut =>
+
+      dut.reset.poke(true.B)
+      dut.io.out.expect(1.U)
+    }
+  }
+  "reset a register works after register has been altered" in {
+    test(new NativeAsyncResetRegModule(1)).withAnnotations(annotations) { dut =>
+      // The register starts at 1 after default reset in tester startup
+      dut.io.out.expect(1.U)
+
+      // register is still zero, after it has been poked to 0
+      dut.io.in.poke(1.U)
+      dut.io.out.expect(1.U)
+
+      // after clock is stepped, now poked input from above appears as output
+      dut.clock.step()
+      dut.io.out.expect(1.U)
+
+      // register is set back to zero
+      dut.io.in.poke(0.U)
+      dut.io.out.expect(1.U)
+      dut.clock.step()
+      dut.io.out.expect(0.U)
+
+      // reset of register shows up immediately
+      dut.reset.poke(true.B)
+      dut.io.out.expect(1.U)
+
+      // after de-assert of reset, register retains value
+      dut.reset.poke(false.B)
+      dut.io.out.expect(1.U)
+
+      // after step zero from io.in is now seen in register
+      dut.clock.step()
+      dut.io.out.expect(0.U)
+    }
+  }
+  "de-assert reset behaviour" in {
+    test(new NativeAsyncResetRegModule(1)).withAnnotations(annotations) { dut =>
+      // register is reset at startup, and set to zero by poking
+      dut.clock.step()
+      dut.io.out.expect(0.U)
+      dut.io.in.poke(0.U)
+      dut.clock.step()
+      dut.io.out.expect(0.U)
+
+      // reset of register shows up immediately
+      dut.reset.poke(true.B)
+      dut.io.out.expect(1.U)
+
+      // after de-assert of reset, register retains value
+      dut.reset.poke(false.B)
+      dut.io.out.expect(1.U)
+
+      dut.clock.step()
+      dut.io.out.expect(0.U)
+
+    }
+  }
+  "feedback into itself" in {
+    test(new NativeAsyncResetFeedbackModule).withAnnotations(annotations) { dut =>
+      dut.io.out(0).expect(0.U)
+      dut.io.out(1).expect(1.U)
+
+      dut.clock.step()
+
+      dut.io.out(0).expect(1.U)
+      dut.io.out(1).expect(0.U)
+
+      dut.clock.step()
+
+      dut.io.out(0).expect(0.U)
+      dut.io.out(1).expect(1.U)
+
+      dut.reset.poke(true.B)
+
+      dut.io.out(0).expect(0.U)
```

Comment:
Maybe you should poke reset on a cycle where you expect change from the registers?

Reply:
done